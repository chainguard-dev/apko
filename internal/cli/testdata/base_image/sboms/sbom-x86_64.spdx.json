{
  "SPDXID": "SPDXRef-DOCUMENT",
  "name": "sbom-sha256:bf74ddaf55d32ec9672a0a40efc6cb1bf0a167763c18fc22586c8a301167822f",
  "spdxVersion": "SPDX-2.3",
  "creationInfo": {
    "created": "1970-01-01T00:00:00Z",
    "creators": [
      "Tool: apko (0.13.2)",
      "Organization: Chainguard, Inc"
    ],
    "licenseListVersion": "3.16"
  },
  "dataLicense": "CC0-1.0",
  "documentNamespace": "https://spdx.org/spdxdocs/apko/",
  "documentDescribes": [
    "SPDXRef-Package-sha256-2ef91a9967f2e1759ea49a8c01cf6a45dd9f9af71fe09bcf2b86175bc4a71314"
  ],
  "files": [],
  "packages": [
    {
      "SPDXID": "SPDXRef-Package-sha256-2ef91a9967f2e1759ea49a8c01cf6a45dd9f9af71fe09bcf2b86175bc4a71314",
      "name": "sha256:2ef91a9967f2e1759ea49a8c01cf6a45dd9f9af71fe09bcf2b86175bc4a71314",
      "versionInfo": "sha256:2ef91a9967f2e1759ea49a8c01cf6a45dd9f9af71fe09bcf2b86175bc4a71314",
      "filesAnalyzed": false,
      "description": "apko container image",
      "downloadLocation": "NOASSERTION",
      "supplier": "Organization: Pretend",
      "primaryPackagePurpose": "CONTAINER",
      "checksums": [
        {
          "algorithm": "SHA256",
          "checksumValue": "2ef91a9967f2e1759ea49a8c01cf6a45dd9f9af71fe09bcf2b86175bc4a71314"
        }
      ],
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceLocator": "pkg:oci/golden@sha256%3A2ef91a9967f2e1759ea49a8c01cf6a45dd9f9af71fe09bcf2b86175bc4a71314?arch=amd64\u0026mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson\u0026os=linux",
          "referenceType": "purl"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-Package-sha256-bf74ddaf55d32ec9672a0a40efc6cb1bf0a167763c18fc22586c8a301167822f",
      "name": "sha256:bf74ddaf55d32ec9672a0a40efc6cb1bf0a167763c18fc22586c8a301167822f",
      "versionInfo": "1.0.0",
      "filesAnalyzed": false,
      "description": "apko operating system layer",
      "downloadLocation": "NOASSERTION",
      "supplier": "Organization: Pretend",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceLocator": "pkg:oci/golden@sha256%3Abf74ddaf55d32ec9672a0a40efc6cb1bf0a167763c18fc22586c8a301167822f?arch=amd64\u0026mediaType=application%2Fvnd.oci.image.layer.v1.tar%2Bgzip\u0026os=linux",
          "referenceType": "purl"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-Package-pretend-baselayout-1.0.0-r0",
      "name": "pretend-baselayout",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": true,
      "hasFiles": [],
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/pretend-baselayout@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ],
      "packageVerificationCode": {
        "packageVerificationCodeValue": "8058dd7b970804f7b4a4f6e10374b4d02a5a01f3"
      }
    },
    {
      "SPDXID": "SPDXRef-Package-package-x-1.0.0-r0",
      "name": "package-x",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": true,
      "hasFiles": [],
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/package-x@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ],
      "packageVerificationCode": {
        "packageVerificationCodeValue": "162018767ec8a51a6d1e6c2841be6f55724dec80"
      }
    },
    {
      "SPDXID": "SPDXRef-Package-package-y-1.0.0-r0",
      "name": "package-y",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": true,
      "hasFiles": [],
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/package-y@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ],
      "packageVerificationCode": {
        "packageVerificationCodeValue": "162018767ec8a51a6d1e6c2841be6f55724dec80"
      }
    }
  ],
  "relationships": [
    {
      "spdxElementId": "SPDXRef-Package-sha256-2ef91a9967f2e1759ea49a8c01cf6a45dd9f9af71fe09bcf2b86175bc4a71314",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-Package-sha256-bf74ddaf55d32ec9672a0a40efc6cb1bf0a167763c18fc22586c8a301167822f"
    },
    {
      "spdxElementId": "SPDXRef-Package-sha256-bf74ddaf55d32ec9672a0a40efc6cb1bf0a167763c18fc22586c8a301167822f",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-Package-pretend-baselayout-1.0.0-r0"
    },
    {
      "spdxElementId": "SPDXRef-Package-sha256-bf74ddaf55d32ec9672a0a40efc6cb1bf0a167763c18fc22586c8a301167822f",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-Package-package-x-1.0.0-r0"
    },
    {
      "spdxElementId": "SPDXRef-Package-sha256-bf74ddaf55d32ec9672a0a40efc6cb1bf0a167763c18fc22586c8a301167822f",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-Package-package-y-1.0.0-r0"
    },
  ]
}
