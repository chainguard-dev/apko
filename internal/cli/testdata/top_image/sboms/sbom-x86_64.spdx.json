{
  "SPDXID": "SPDXRef-DOCUMENT",
  "name": "sbom-sha256:440b16a8805032601569f324b32f5a41f64761e02fca0dcb0a0382c498de87e1",
  "spdxVersion": "SPDX-2.3",
  "creationInfo": {
    "created": "1970-01-01T00:00:00Z",
    "creators": [
      "Tool: apko (devel)",
      "Organization: Chainguard, Inc"
    ],
    "licenseListVersion": "3.16"
  },
  "dataLicense": "CC0-1.0",
  "documentNamespace": "https://spdx.org/spdxdocs/apko/",
  "documentDescribes": [
    "SPDXRef-Package-sha256-1f6a8feb0923498f610445de79658aa551053e55aa36b1978858309f6214d0da"
  ],
  "files": [
    {
      "SPDXID": "SPDXRef-File--etc-os-release",
      "fileName": "etc/os-release",
      "licenseConcluded": "NOASSERTION",
      "checksums": [
        {
          "algorithm": "SHA1",
          "checksumValue": "ca5e527bbb8a5cc9c4c2d2b4e29618d8ca3be5f8"
        },
        {
          "algorithm": "SHA256",
          "checksumValue": "53db9a78bf280b687690cdda8e275157713d725aaa54fe662f3bcf2f51eea457"
        },
        {
          "algorithm": "SHA512",
          "checksumValue": "ab0cf3e8797ea0e989960f40808333150b4c1ee2fa056b1cc271a98da0e1e8638bbaf5e89aad91436fb3b994a74b1886714df6b153125e712b08d3eaacd24dd5"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-File--x-hello.txt",
      "fileName": "x/hello.txt",
      "licenseConcluded": "NOASSERTION",
      "checksums": [
        {
          "algorithm": "SHA1",
          "checksumValue": "a8eec30a5b2d71bc890175f5b361ebb28d7c54a8"
        },
        {
          "algorithm": "SHA256",
          "checksumValue": "3b09aeb6f5f5336beb205d7f720371bc927cd46c21922e334d47ba264acb5ba4"
        },
        {
          "algorithm": "SHA512",
          "checksumValue": "dec5b5e130d1694e65b1bf3f915024d51e87817248ab625e8732e183c321a9aaa09f92c04ed3d1d3a5b173838bd40ff5b1c8bb6318bcea70f4f72a8bff0ec2a1"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-File--y-hello.txt",
      "fileName": "y/hello.txt",
      "licenseConcluded": "NOASSERTION",
      "checksums": [
        {
          "algorithm": "SHA1",
          "checksumValue": "a8eec30a5b2d71bc890175f5b361ebb28d7c54a8"
        },
        {
          "algorithm": "SHA256",
          "checksumValue": "3b09aeb6f5f5336beb205d7f720371bc927cd46c21922e334d47ba264acb5ba4"
        },
        {
          "algorithm": "SHA512",
          "checksumValue": "dec5b5e130d1694e65b1bf3f915024d51e87817248ab625e8732e183c321a9aaa09f92c04ed3d1d3a5b173838bd40ff5b1c8bb6318bcea70f4f72a8bff0ec2a1"
        }
      ]
    }
  ],
  "packages": [
    {
      "SPDXID": "SPDXRef-Package-sha256-1f6a8feb0923498f610445de79658aa551053e55aa36b1978858309f6214d0da",
      "name": "sha256:1f6a8feb0923498f610445de79658aa551053e55aa36b1978858309f6214d0da",
      "versionInfo": "sha256:1f6a8feb0923498f610445de79658aa551053e55aa36b1978858309f6214d0da",
      "filesAnalyzed": false,
      "description": "apko container image",
      "downloadLocation": "NOASSERTION",
      "supplier": "Organization: Replaces",
      "primaryPackagePurpose": "CONTAINER",
      "checksums": [
        {
          "algorithm": "SHA256",
          "checksumValue": "1f6a8feb0923498f610445de79658aa551053e55aa36b1978858309f6214d0da"
        }
      ],
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceLocator": "pkg:oci/golden@sha256%3A1f6a8feb0923498f610445de79658aa551053e55aa36b1978858309f6214d0da?arch=amd64\u0026mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson\u0026os=linux",
          "referenceType": "purl"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-Package-sha256-440b16a8805032601569f324b32f5a41f64761e02fca0dcb0a0382c498de87e1",
      "name": "sha256:440b16a8805032601569f324b32f5a41f64761e02fca0dcb0a0382c498de87e1",
      "versionInfo": "1.0.0",
      "filesAnalyzed": false,
      "description": "apko operating system layer",
      "downloadLocation": "NOASSERTION",
      "supplier": "Organization: Replaces",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceLocator": "pkg:oci/golden@sha256%3A440b16a8805032601569f324b32f5a41f64761e02fca0dcb0a0382c498de87e1?arch=amd64\u0026mediaType=application%2Fvnd.oci.image.layer.v1.tar%2Bgzip\u0026os=linux",
          "referenceType": "purl"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-Package-replayout-1.0.0-r0",
      "name": "replayout",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": true,
      "hasFiles": [
        "SPDXRef-File--etc-os-release"
      ],
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/replayout@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ],
      "packageVerificationCode": {
        "packageVerificationCodeValue": "d70ccb53a479f44c7ac6f023492b3520f23c21b3"
      }
    },
    {
      "SPDXID": "SPDXRef-Package-pretend-baselayout-1.0.0-r0",
      "name": "pretend-baselayout",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": true,
      "hasFiles": [
        "SPDXRef-File--etc-os-release"
      ],
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/pretend-baselayout@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ],
      "packageVerificationCode": {
        "packageVerificationCodeValue": "8058dd7b970804f7b4a4f6e10374b4d02a5a01f3"
      }
    },
    {
      "SPDXID": "SPDXRef-Package-package-x-1.0.0-r0",
      "name": "package-x",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": true,
      "hasFiles": [
        "SPDXRef-File--x-hello.txt"
      ],
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/package-x@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ],
      "packageVerificationCode": {
        "packageVerificationCodeValue": "162018767ec8a51a6d1e6c2841be6f55724dec80"
      }
    },
    {
      "SPDXID": "SPDXRef-Package-package-y-1.0.0-r0",
      "name": "package-y",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": true,
      "hasFiles": [
        "SPDXRef-File--y-hello.txt"
      ],
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/package-y@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ],
      "packageVerificationCode": {
        "packageVerificationCodeValue": "162018767ec8a51a6d1e6c2841be6f55724dec80"
      }
    }
  ],
  "relationships": [
    {
      "spdxElementId": "SPDXRef-Package-sha256-1f6a8feb0923498f610445de79658aa551053e55aa36b1978858309f6214d0da",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-Package-sha256-440b16a8805032601569f324b32f5a41f64761e02fca0dcb0a0382c498de87e1"
    },
    {
      "spdxElementId": "SPDXRef-Package-sha256-440b16a8805032601569f324b32f5a41f64761e02fca0dcb0a0382c498de87e1",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-Package-replayout-1.0.0-r0"
    },
    {
      "spdxElementId": "SPDXRef-Package-replayout-1.0.0-r0",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-File--etc-os-release"
    },
    {
      "spdxElementId": "SPDXRef-Package-pretend-baselayout-1.0.0-r0",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-File--etc-os-release"
    },
    {
      "spdxElementId": "SPDXRef-Package-package-x-1.0.0-r0",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-File--x-hello.txt"
    },
    {
      "spdxElementId": "SPDXRef-Package-package-y-1.0.0-r0",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-File--y-hello.txt"
    }
  ]
}
