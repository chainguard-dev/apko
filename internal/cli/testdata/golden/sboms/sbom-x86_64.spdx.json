{
  "SPDXID": "SPDXRef-DOCUMENT",
  "name": "sbom-sha256:89c9e3bd4d4968247bdf82182560718f22427628a3c112902377fa7ff897f9f7",
  "spdxVersion": "SPDX-2.3",
  "creationInfo": {
    "created": "1970-01-01T00:00:00Z",
    "creators": [
      "Tool: apko (devel)",
      "Organization: Chainguard, Inc"
    ],
    "licenseListVersion": "3.16"
  },
  "dataLicense": "CC0-1.0",
  "documentNamespace": "https://spdx.org/spdxdocs/apko/",
  "documentDescribes": [
    "SPDXRef-Package-sha256-0c3a3431cb3d08f1de9b3ebd19671fe54c227a6cfd33c1164cb8f67a86baadea"
  ],
  "packages": [
    {
      "SPDXID": "SPDXRef-Package-sha256-0c3a3431cb3d08f1de9b3ebd19671fe54c227a6cfd33c1164cb8f67a86baadea",
      "name": "sha256:0c3a3431cb3d08f1de9b3ebd19671fe54c227a6cfd33c1164cb8f67a86baadea",
      "versionInfo": "sha256:0c3a3431cb3d08f1de9b3ebd19671fe54c227a6cfd33c1164cb8f67a86baadea",
      "filesAnalyzed": false,
      "description": "apko container image",
      "downloadLocation": "NOASSERTION",
      "supplier": "Organization: Replaces",
      "primaryPackagePurpose": "CONTAINER",
      "checksums": [
        {
          "algorithm": "SHA256",
          "checksumValue": "0c3a3431cb3d08f1de9b3ebd19671fe54c227a6cfd33c1164cb8f67a86baadea"
        }
      ],
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceLocator": "pkg:oci/golden@sha256%3A0c3a3431cb3d08f1de9b3ebd19671fe54c227a6cfd33c1164cb8f67a86baadea?arch=amd64\u0026mediaType=application%2Fvnd.oci.image.manifest.v1%2Bjson\u0026os=linux",
          "referenceType": "purl"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-Package-sha256-89c9e3bd4d4968247bdf82182560718f22427628a3c112902377fa7ff897f9f7",
      "name": "sha256:89c9e3bd4d4968247bdf82182560718f22427628a3c112902377fa7ff897f9f7",
      "versionInfo": "1.0.0",
      "filesAnalyzed": false,
      "description": "apko operating system layer",
      "downloadLocation": "NOASSERTION",
      "supplier": "Organization: Replaces",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE-MANAGER",
          "referenceLocator": "pkg:oci/golden@sha256%3A89c9e3bd4d4968247bdf82182560718f22427628a3c112902377fa7ff897f9f7?arch=amd64\u0026mediaType=application%2Fvnd.oci.image.layer.v1.tar%2Bgzip\u0026os=linux",
          "referenceType": "purl"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-Package-pretend-baselayout-1.0.0-r0",
      "name": "pretend-baselayout",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": false,
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "supplier": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/pretend-baselayout@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ]
    },
    {
      "SPDXID": "SPDXRef-Package-replayout-1.0.0-r0",
      "name": "replayout",
      "versionInfo": "1.0.0-r0",
      "filesAnalyzed": false,
      "licenseConcluded": "NOASSERTION",
      "licenseDeclared": "MIT",
      "downloadLocation": "NOASSERTION",
      "originator": "Organization: Unknown",
      "supplier": "Organization: Unknown",
      "copyrightText": "\n",
      "externalRefs": [
        {
          "referenceCategory": "PACKAGE_MANAGER",
          "referenceLocator": "pkg:apk/unknown/replayout@1.0.0-r0?arch=x86_64",
          "referenceType": "purl"
        }
      ]
    }
  ],
  "relationships": [
    {
      "spdxElementId": "SPDXRef-Package-sha256-0c3a3431cb3d08f1de9b3ebd19671fe54c227a6cfd33c1164cb8f67a86baadea",
      "relationshipType": "CONTAINS",
      "relatedSpdxElement": "SPDXRef-Package-sha256-89c9e3bd4d4968247bdf82182560718f22427628a3c112902377fa7ff897f9f7"
    }
  ]
}
